client/src/components/graphs/OrganizationalChart.tsx

```diff
@@ -189,260 +158,669 @@ const OrganizationalChart = ({
     pageDispatchers
   })
 
-  const canvas = d3.select(canvasRef.current)
-  const link = d3.select(linkRef.current)
-  const node = d3.select(nodeRef.current)
-  const attachmentsEnabled = !Settings.fields.attachment.featureDisabled
+  if (done) {
+    return result
+  }
 
-  useEffect(() => {
-    data &&
-      setRoot(
-        d3.hierarchy(data.organization, d =>
-          expanded.includes(d.uuid)
-            ? data.organization.descendantOrgs.filter(
-              org => org.parentOrg?.uuid === d.uuid
-            )
-            : null
-        )
-      )
-  }, [data, expanded])
+  const organization = data?.organization
+  if (!organization) {
+    return <p>Loading...</p>
+  }
 
-  useEffect(() => {
-    if (!data || !root) {
+  return (
+    <ReactFlowProvider>
+      <OrganizationFlowChart
+        organization={organization}
+        exportTitle={exportTitle}
+        width={width}
+      />
+    </ReactFlowProvider>
+  )
+}
+
+interface OrganizationFlowChartProps {
+  organization: { descendantOrgs: any[]; ascendantOrgs: any[] }
+  width?: number
+  exportTitle?: string
+}
+
+const OrganizationFlowChart = ({
+  organization,
+  width,
+  exportTitle
+}: OrganizationFlowChartProps) => {
+  const [showApp6Symbols, setShowApp6Symbols] = useState(false)
+  const [depthLimit, setDepthLimit] = useState(3)
+  const [maxDepth, setMaxDepth] = useState(0)
+  const [positionsFilter, setPositionFilter] =
```

Comment:
value and setter still don't match (useState hook should always be of the form `const [someValue, setSomeValue] = useState…` )